### DISCUSSION TITLE
Migrate Ansible to Docker (need help)

### DISCUSSION THREAD

**Discussion by burnscm** [AzuraCast/AzuraCast#6221] [General]
cd /var/azuracast/www
bash update.sh
bash docker-migrate.sh

But, i can't get/find   docker-migrate.sh.

I have an ansible installation (installation 2021 / updated)  and must migrate to docker. I can't setup from scratch, (to many configuration, playlists and users...)

where can i find a howto?

I have a new (empty) docker installation, but i can't restore a 'ansible' backup.

Regards

**Comment by BusterNeece** [comment]
@burnscm You actually should be able to restore an Ansible backup in a Docker installation. That's our recommended way of moving from Ansible to the Docker installation format.

**Comment by burnscm** [comment]
NO! does not work.
i do only the DB
./docker.sh   restore  db.tar.gz

Reload System Data
------------------


                                                                                                                        
 [OK] Local cache flushed.                                                                                              
                                                                                                                        

[2023-04-21T10:17:25.440583+00:00] AzuraCast.ERROR: Doctrine\DBAL\Exception\InvalidFieldNameException: An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 't0.dropbox_app_key' in 'field list' (uncaught exception) at /var/azuracast/www/vendor/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php line 67 while running console command `azuracast:restore` {"exception":"[object] (Doctrine\\DBAL\\Exception\\InvalidFieldNameException(code: 1054): An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 't0.dropbox_app_key' in 'field list' at /var/azuracast/www/vendor/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php:67)\n[previous exception] [object] (Doctrine\\DBAL\\Driver\\PDO\\Exception(code: 1054): SQLSTATE[42S22]: Column not found: 1054 Unknown column 't0.dropbox_app_key' in 'field list' at /var/azuracast/www/vendor/doctrine/dbal/src/Driver/PDO/Exception.php:28)\n[previous exception] [object] (PDOException(code: 42S22): SQLSTATE[42S22]: Column not found: 1054 Unknown column 't0.dropbox_app_key' in 'field list' at /var/azuracast/www/vendor/doctrine/dbal/src/Driver/PDO/Statement.php:121)"} []
[2023-04-21T10:17:25.441804+00:00] AzuraCast.WARNING: Console command `azuracast:restore` exited with error code 1054. [] []

In ExceptionConverter.php line 67:
                                                                                                                                              
  An exception occurred while executing a query: SQLSTATE[42S22]: Column not found: 1054 Unknown column 't0.dropbox_app_key' in 'field list'  
                                                                                                                                              

In Exception.php line 28:
                                                                                               
  SQLSTATE[42S22]: Column not found: 1054 Unknown column 't0.dropbox_app_key' in 'field list'  
                                                                                               

In Statement.php line 121:
                                                                                               
  SQLSTATE[42S22]: Column not found: 1054 Unknown column 't0.dropbox_app_key' in 'field list'  
                                                                                               

azuracast:restore [--restore] [--release] [--] [<path>]

[+] Running 1/1
 ✔ Network azuracast_default  Removed                                                                                                             0.4s 
WARN[0000] volume "azuracast_backups" already exists but was not created by Docker Compose. Use `external: true` to use an existing volume 
WARN[0000] volume "azuracast_station_data" already exists but was not created by Docker Compose. Use `external: true` to use an existing volume 
[+] Running 3/3
 ✔ Network azuracast_default    Created                                                                                                           0.1s 
 ✔ Container azuracast          Started                                                                                                           3.0s 
 ✔ Container azuracast_updater  Started                                                                                                           0.4s 
 
 ...500 bad gateway
 ................................................................................................................................................................................................
 
 ./docker.sh restore-legacy db.tar.gz
 
 root@azuracast2:/var/azuracast# ./docker.sh restore-legacy db.tar.gz
[+] Running 3/3
 ✔ Container azuracast          Removed                                                                                                                                13.9s 
 ✔ Container azuracast_updater  Removed                                                                                                                                 0.3s 
 ✔ Network azuracast_default    Removed                                                                                                                                 0.4s 
azuracast_db_data
azuracast_station_data
azuracast_db_data
azuracast_station_data
tmp/azuracast_backup_mariadb/db.sql
WARN[0000] volume "azuracast_db_data" already exists but was not created by Docker Compose. Use `external: true` to use an existing volume 
WARN[0000] volume "azuracast_station_data" already exists but was not created by Docker Compose. Use `external: true` to use an existing volume 
WARN[0000] volume "azuracast_backups" already exists but was not created by Docker Compose. Use `external: true` to use an existing volume 
[+] Running 3/3
 ✔ Network azuracast_default    Created                                                                                                                                 0.1s 
 ✔ Container azuracast_updater  Started                                                                                                                                 0.6s 
 ✔ Container azuracast          Started                                                                                                                                 3.1s 
root@azuracast2:/var/azuracast# 

leaves an empty installation
